relay-server/src/utils/metrics_rate_limits.rs

```diff
@@ -94,25 +94,31 @@ where
     quantities
 }
 
-fn report_rejected_metrics<I: IntoIterator<Item = Bucket>>(
+fn report_rejected_metrics<'a, I, V>(
     metric_stats: &MetricStats,
     scoping: Scoping,
     buckets: I,
     outcome: Outcome,
-) {
+) where
+    I: IntoIterator<Item = V>,
+    V: Into<BucketView<'a>>,
+{
     for bucket in buckets {
         metric_stats.track_metric(scoping, bucket, outcome.clone())
     }
 }
 
-pub fn reject_metrics<I: IntoIterator<Item = Bucket>>(
+pub fn reject_metrics<'a, I, V>(
     addr: &Addr<TrackOutcome>,
     quantities: SourceQuantities,
     scoping: Scoping,
     outcome: Outcome,
     metric_stats: Option<&MetricStats>,
     buckets: Option<I>,
```

Comment:
```suggestion
pub fn reject_metrics<'a, T: Into<BucketView<'a>>>(
    addr: &Addr<TrackOutcome>,
    quantities: SourceQuantities,
    scoping: Scoping,
    outcome: Outcome,
    metric_stats: Option<&MetricStats>,
    buckets: impl IntoIterator<Item = T>,
```

This should work and means you only have to supply a single generic. The trick is that `None` implements `IntoIterator`.

Reply:
The problem here is that the iterator will return one element or zero of type `T` which in our case would become `Vec<Bucket>` which doesn't implement `From` for `BucketView`.

Reply:
I am not sure I understand, I was thinking of something like this:

```rs

fn foo<T: Into<u32>>(_: impl IntoIterator<Item = T>) {}

fn main() {
    foo::<u16>(None);
    foo(vec![1u16]);
}
```